src/Microsoft.Extensions.Configuration/ConfigurationBuilderExtensions.cs

```diff
@@ -7,9 +7,42 @@
 
 namespace Microsoft.Extensions.Configuration
 {
-    public static class MemoryConfigurationExtensions
+    public static class ConfigurationBuilderExtensions
     {
         /// <summary>
+        /// Includes an existing IConfiguration as a configuration provider to <paramref name="configuraton"/>.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="config">The <see cref="IConfiguration"/> to include.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddConfiguration(this IConfigurationBuilder configurationBuilder, IConfiguration configuration)
```

Comment:
Having `configurationBuilder.AddConfiguration()` is a bit puzzling IMO. I would suggest using a name that conveys the chainedness, e.g. `AddChainedConfiguration()`.